### ISSUE TITLE
[BUG] Infinite Looping in Event Activity in v3

### ISSUE THREAD

**Issue by sagargolu** [elsa-workflows/elsa-core#5409]
## Description
We have a workflow with events 1, 2, 3, and 4.
The execution sequence 1 -> 2 -> 3 works as expected. However, once event 3 is completed, event 4 is triggered, which loops back to event 3. This is where the problem starts. After event 4 is completed, the workflow enters an infinite loop, continuously cycling through events 2, 3, and 4 without waiting for user intervention.
The workflow repeats in the following manner: 2 -> 3 -> 4 -> 2 -> 3 -> 4 -> 2 -> 3 -> 4, and so on. This looping happens without pausing to allow the user to manually complete the events.
Ideally, the execution should wait for the user to manually complete each event, even if the event has been completed before and is triggered again. The workflow should only proceed once the user has manually completed the event.
This is a minimum reproducible example. In a real-world scenario, there would be an if block before triggering event 4.

## Steps to Reproduce
To help us identify the issue more quickly, please follow these guidelines:

1. **Detailed Steps**: I am attaching video and json of that workflow so that you will have some idea about it.

3. **Attachments**:
   - **Workflow JSON**: 
[workflow-2.json](https://github.com/elsa-workflows/elsa-core/files/15347058/workflow-2.json)

4. **Video/Screenshots**: 

https://github.com/elsa-workflows/elsa-core/assets/79979623/25be56cf-a3cd-4ee4-aa19-8478cc00185a

Providing a comprehensive and clear description will greatly assist in diagnosing the problem quickly and accurately. If you're unable to share some of the requested information publicly, please note that, and we can arrange for secure information transfer.

## Expected Behavior
Workflow should not continue further automatically when it returns to the Event2. It should wait to complete the Event2. This flow was working perfectly in v2 of elsa.

## Actual Behavior
You can see it in the uploaded video it is autocompleting Event2 /Event3 and going into infinite loop.

**Comment by NrujalSharma** [comment]
Is there any update on this @sfmskywalker?
I am facing the same issue

**Comment by sfmskywalker** [comment]
Although I have yet to look into this issue specifically, I’m pretty certain that, based on the description, it will be a non-issue after the refactoring I’m currently performing, which includes a significant change in the way events are processed.

**Comment by sfmskywalker** [comment]
I can confirm that the issue exists in the current main branch, but that it has been fixed with the feature branch that I mentioned. Once that gets merged, I will close this issue as fixed.

**Comment by NrujalSharma** [comment]
Sounds good. Thanks for the update!
Looking forward to the release.

**Comment by sagargolu** [comment]
When is this expected to get it merged into main branch?

**Comment by sfmskywalker** [comment]
I’m aiming for sometime next week, contingent on progress on some remaining tasks and bugs.

**Comment by sagargolu** [comment]
@sfmskywalker ,

Thanks for doing this on Priority.
